vapor: init at 4.105.0

fbettag opened the pull request:
## Summary
- Add Vapor, a server-side Swift HTTP web framework at version 4.105.0
- Package provides Vapor framework source code for Swift Package Manager integration
- Includes auto-update script for maintaining latest version
- Compatible with existing Swift 5.8 toolchain in nixpkgs

## Test plan
- [x] Package builds successfully with `nix-build -A vapor`
- [x] Sources are properly installed in `/nix/store/.../share/vapor/`
- [x] Package.swift manifest is available for Swift Package Manager
- [x] Auto-update script is functional
- [x] Follows nixpkgs by-name structure conventions

fbettag committed:
vapor: init at 4.105.0

Add Vapor, a server-side Swift HTTP web framework. This package provides
the Vapor framework source code for use with Swift Package Manager, making
it available for Swift projects within the Nix ecosystem.

The package includes:
- Complete Vapor framework source code
- Package.swift manifest for Swift Package Manager integration
- Auto-update script for maintaining the latest version

fbettag committed:
vapor: fix formatting issues

Remove trailing whitespace to satisfy treefmt checks

JohnRTitor commented on pkgs/by-name/va/vapor/package.nix:
```diff
@@ -0,0 +1,58 @@
+  nix-update-script,
+}:
+
+stdenv.mkDerivation rec {
```
Please use finalAttrs.

JohnRTitor commented on pkgs/by-name/va/vapor/package.nix:
```diff
@@ -0,0 +1,58 @@
+
+  passthru.updateScript = nix-update-script { attrPath = pname; };
+
+  meta = with lib; {
```
with lib is not recommended anymore. So please use `lib.` properties explictly.

fbettag committed:
vapor: address review feedback

- Convert to finalAttrs pattern as requested by @JohnRTitor
- Replace 'with lib' with explicit lib. prefixes
- Use finalAttrs.version and finalAttrs.pname for proper overriding support
- Maintain all existing functionality and formatting compliance

Addresses: https://github.com/NixOS/nixpkgs/pull/440549#discussion_r2328363645
Addresses: https://github.com/NixOS/nixpkgs/pull/440549#discussion_r2328363810

fbettag commented:
@JohnRTitor thanks for the review. I should have addressed all concerns now.

fbettag committed:
vapor: adds fbettag as maintainer.

fbettag committed:
vapor: fix final with lib usage in maintainers

Remove last remaining 'with lib.maintainers' usage to fully comply
with the 'no with lib' guideline from review feedback.

This should resolve the treefmt lint failure.

Addresses: https://github.com/NixOS/nixpkgs/pull/440549#discussion_r2328363810

fbettag committed:
maintainers: fix alphabetical order for fbettag

The keep-sorted formatter requires fbettag to come after fbergroth
alphabetically. This fixes the treefmt CI failure.

Required for: vapor package addition in pkgs/by-name/va/vapor/

fbettag committed:
vapor: update to 4.115.1 and use tag pattern

- Update to latest release 4.115.1 as suggested by @JohnRTitor
- Change from rev hash to tag = finalAttrs.version pattern
- Update hash for new version

Addresses: https://github.com/NixOS/nixpkgs/pull/440549#discussion_r2329466058
